gpt_json/tests/test_streaming.py

```diff
@@ -0,0 +1,268 @@
+import json
+from typing import Any, List
+from unittest.mock import MagicMock, patch
+
+import openai
+import pytest
+import tiktoken
+from pydantic import BaseModel
+
+from gpt_json.gpt import GPTJSON
+from gpt_json.models import GPTMessage, GPTMessageRole, GPTModelVersion
+from gpt_json.streaming import StreamEventEnum, StreamingObject
+
+enc = tiktoken.get_encoding("cl100k_base")
+
+def _tokenize(text):
+    return [enc.decode([tok]) for tok in enc.encode(text)]
+
+def _tuple_merge(t1, t2):
+    t1 = (t1,) if not isinstance(t1, tuple) else t1
+    t2 = (t2,) if not isinstance(t2, tuple) else t2
+
+    return t1 + t2
+
+def _get_expected_stream_partial_objs(full_object):
+    """This function implements the semantics of the behavior we expect from GPTJSON.stream().
```

Comment:
This function's getting a bit long, mind refactoring into a separate class:

I'm thinking something like:

```
class ExpectedPartialObjectStreamHarness:
    def __call__(self, full_obj: Any):
        if isinstance(full_obj, list):
            yield from self.handle_list(full_obj)
        elif isinstance(full_obj, dict):
            yield from self.handle_dict(full_obj)

    def handle_list(self, full_obj):
        value_iterators = [self(v) for v in full_object]
        ...
```

I imagine this might also provide better typehinting support.

Reply:
oh hell ya, nice